src/pt/portugamangas/src/eu/kanade/tachiyomi/extension/pt/portugamangas/PortugaMangas.kt

```diff
@@ -0,0 +1,182 @@
+import eu.kanade.tachiyomi.source.model.UpdateStrategy
+import eu.kanade.tachiyomi.source.online.ParsedHttpSource
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.Response
+import org.jsoup.Jsoup
+import org.jsoup.nodes.Document
+import org.jsoup.nodes.Element
+import java.text.SimpleDateFormat
+import java.util.Locale
+import java.util.concurrent.TimeUnit
+
+class PortugaMangas : ParsedHttpSource() {
+
+    override val name = "Portuga Mangas"
+
+    override val baseUrl = "https://portugamanga.online"
+
+    override val lang = "pt-BR"
+
+    override val supportsLatest = true
+
+    override val client: OkHttpClient = network.cloudflareClient.newBuilder()
+        .rateLimitHost(baseUrl.toHttpUrl(), 1, 2, TimeUnit.SECONDS)
+        .build()
+
+    override fun popularMangaRequest(page: Int): Request = GET(baseUrl, headers)
+
+    override fun popularMangaSelector(): String = "div#maisLidos > div.itemmanga"
+
+    override fun popularMangaFromElement(element: Element): SManga = SManga.create().apply {
+        title = element.selectFirst("h3")?.text() ?: "Title not found"
+        thumbnail_url = element.selectFirst("img")!!.srcAttr()
+        setUrlWithoutDomain(element.attr("href"))
+    }
+
+    override fun popularMangaNextPageSelector(): String? = null
+
+    override fun latestUpdatesRequest(page: Int): Request {
+        val url = "$baseUrl".toHttpUrl().newBuilder()
+            .addQueryParameter("pagina", "$page")
+            .build()
+        return GET(url, headers)
+    }
+
+    override fun latestUpdatesSelector() = "div.atualizacao"
+
+    override fun latestUpdatesFromElement(element: Element): SManga = SManga.create().apply {
+        title = element.selectFirst("a > h3")?.text() ?: "Title not found"
+        thumbnail_url = element.selectFirst("div > img")!!.absUrl("src")
+        setUrlWithoutDomain(element.selectFirst("a")!!.attr("href"))
+    }
+
+    override fun latestUpdatesNextPageSelector(): String? = "ul.pagination [aria-label=Next]"
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        val url = "$baseUrl/mangas".toHttpUrl().newBuilder()
+            .addQueryParameter("busca", query)
+            .addQueryParameter("pagina", "$page")
+            .build()
+        return GET(url, headers)
+    }
+
+    private fun hasNextPage(currentPage: Int, document: Document): Boolean {
+        val pageAmount = document.select("ul.pagination li:not([aria-hidden])").size
+        return pageAmount != currentPage && currentPage != -1
+    }
+
+    private fun searchMangaParse(document: Document): List<SManga> =
+        document.select(searchMangaSelector()).map { this.searchMangaFromElement(it) }
+
+    override fun searchMangaParse(response: Response): MangasPage {
+        val currentPage = response.request.url.queryParameter("pagina")?.toInt() ?: -1
+        val document = response.parseAsDocument()
```

Comment:
```suggestion
        val document = response.asJsoup()
```
use app provided `asJsoup` function, also need to import it first.

Reply:
Fix: https://github.com/keiyoushi/extensions-source/pull/471/commits/12af94217a8eaaf10eb52a130a56db373c045148